Deploy cashu.1

callebtc opened the pull request.

callebtc committed:
depoloy cashu

callebtc committed:
add cashu dockerfile

callebtc committed:
depends

callebtc committed:
fix typos

callebtc committed:
Merge branch 'main' into deploy_cashu

callebtc committed:
update deployment

mixmasala committed:
Merge remote-tracking branch 'cloaked/main' into deploy_cashu

mixmasala committed:
genconfig, docker: Add Dockerfile.cashu to docker/Makefile

callebtc committed:
add cashu env vars

mixmasala commented:
while testing this I see server_cashu_wallet fails with

```
masala@cloaked1:~/CloakedNetworkPOC/docker/voting_mixnet$ podman logs server_cashu_wallet
Skipping virtualenv creation, as specified in config file.
Warning: 'cashu' is an entry point defined in pyproject.toml, but it's not installed as a script. You may get improper `sys.argv[0]`.

The support to run uninstalled scripts will be removed in a future release.

Run `poetry install` to resolve and get rid of this message.

INFO:     Started server process [1]
INFO:     Waiting for application startup.
Creating database directory: /root/.cashu/wallet
Generated a new mnemonic. To view it, run "cashu info --mnemonic".
Starting Tor.......................................done
ERROR:    Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 677, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 566, in __aenter__
    await self._router.startup()
  File "/usr/local/lib/python3.9/site-packages/starlette/routing.py", line 654, in startup
    await handler()
  File "/app/cashu/cashu/wallet/api/router.py", line 75, in start_wallet
    await wallet.load_mint()
  File "/app/cashu/cashu/wallet/wallet.py", line 576, in load_mint
    await super()._load_mint(keyset_id)
  File "/app/cashu/cashu/wallet/wallet.py", line 212, in _load_mint
    await self._load_mint_keys(keyset_id)
  File "/app/cashu/cashu/wallet/wallet.py", line 160, in _load_mint_keys
    keyset = await self._get_keys(self.url)
  File "/app/cashu/cashu/wallet/wallet.py", line 96, in wrapper
    return await func(self, *args, **kwargs)
  File "/app/cashu/cashu/wallet/wallet.py", line 251, in _get_keys
    self.raise_on_error(resp)
  File "/app/cashu/cashu/wallet/wallet.py", line 130, in raise_on_error
    resp_dict = resp.json()
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

ERROR:    Application startup failed. Exiting.
```

mixmasala committed:
Merge remote-tracking branch 'cloaked/main' into deploy_cashu.1

mixmasala committed:
docker/Dockerfile.cashu: use python:3-11-bookworm

-slim version of image is missing 'ps' command and doesn't seem to work with docker-compose

mixmasala committed:
cloakedproxy: skip building without gio tag

mixmasala committed:
cloakedproxy: add docker-build-linux, docker-build-windows

mixmasala committed:
core/crypto/cert: Sign current+2 so tests do not fail across epoch boundaries
